x-pack/plugins/security/public/authentication/login/login_page.tsx

```diff
@@ -122,19 +139,25 @@ export class LoginPage extends Component<Props, State> {
       ['loginWelcome__contentDisabledForm']: !loginIsSupported,
     });
 
+    const customLogo = this.state.customBranding?.logo;
+    const logo = customLogo ? (
+      <img src={customLogo} width="40" height="40" alt="logo" />
```

Comment:
Yes, this can be `EuiImage`, will change it. `customBranding` is a service exposed from core. It's read from injected metadata on the client side. We're debating still if only admin users should be able to set it.

Reply:
> Yes, this can be EuiImage, will change it. 

:+1: 

> We're debating still if only admin users should be able to set it.

Got it, would be great to know the outcome. Arbitrary content in the image source isn't a big threat, but external image links might conflict with the CSP settings in the future.

Reply:
The conclusion is that we won't be adding any additional privileges to the Advanced Settings. So anyone with `Stack Management` privileges in Kibana will be able to set custom branding.

Reply:
To clarify further: image source won't be link to the external src, but base64-encoded string of the image data

Reply:
> The conclusion is that we won't be adding any additional privileges to the Advanced Settings. So anyone with Stack Management privileges in Kibana will be able to set custom branding.

Got it, thanks for the update.

> To clarify further: image source won't be link to the external src, but base64-encoded string of the image data

Okay, so we'll need to support `data:` or `blob:` in `img-src` (we don't restrict `img-src` today, but we might in the future).